.github/workflows/check_need_autoupgrade_pr_label.yml

```diff
@@ -0,0 +1,108 @@
+# This workflow allows you to check potential changes to the structure of the database,
+# and to apply the “Needs autoupgrade PR” label if this is the case
+name: Check the label 'Needs autoupgrade PR'
+
+on: pull_request_target
+
+jobs:
+  install_prestashop_and_dump_base:
+    name: Install Prestashop and create database dump for base branch
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Docker build
+        run: docker compose build prestashop-git
+
+      - name: Build dependency
+        run: docker compose run --rm prestashop-git composer install --ansi --prefer-dist --no-interaction --no-progress
+
+      - name: Create base database
+        run: docker compose run --rm mysql mysql -hmysql -uroot -pprestashop -e "CREATE DATABASE presta_base;"
+
+      - name: Install shop
+        run: docker compose run --rm prestashop-git php install-dev/index_cli.php \
+          --step=database --db_server=mysql:3306 --db_name=presta_base --db_user=root --db_password=prestashop --prefix=ps_ --db_clear=1 \
+          --domain=localhost:8001 --firstname="Marc" --lastname="Beier" \
+          --password=Toto123! --email=demo@prestashop.com --language=fr --country=fr \
+          --newsletter=0 --send_email=0 --ssl=0
+
+      - name: Export dump
+        run: docker compose run --rm mysql sh -c "exec mysqldump -hmysql -uroot --no-data --compact -pprestashop presta_base" > dump_base.sql
+
+      - name: Upload dump
+        uses: actions/upload-artifact@v4
+        with:
+          name: dump_base
+          path: |
+            dump_base.sql
```

Comment:
I think we can refacto this part to not duplicate it and juste use reusable workflow with params. What do you think ?

Reply:
Are you thinking of a matrix ? totally possible

Reply:
I thought about using an action at first, but I guess a matrix would probably do the job better

Reply:
Although I wouldn't block the PR just for this so it's up to you